scripts/gufi_find

```diff
@@ -108,86 +108,94 @@ def parse_size(size):
     size = int(size) * filesize[unit]
     return prefix, size
 
-def build_select(args):
-    '''Build the SELECT portion of the queries for the treesummary, summary, and entries tables.'''
-    tsummary = []
-    summary  = []
-    entries  = []
-
-    # select must exist
-    if args.select == None:
-        # args.select = [['id', 'path() || "/" || name', 'type',
-        #                 'inode', 'mode', 'nlink', 'uid', 'gid',
-        #                 'size', 'blksize', 'blocks', 'atime',
-        #                 'mtime', 'ctime', 'linkname', 'xattrs',
-        #                 'crtime', 'ossint1', 'ossint2', 'ossint3',
-        #                 'ossint4', 'osstext1', 'osstext2', 'pinode']]
-        if args.type != 'd':
-            args.select = [['path() || "/" || name']]
-        else:
-            args.select = [['path()']]
+def build_init(args):
+    '''Build the CREATE TABLE query'''
+    init = ['name TEXT', 'type TEXT']
+
+    if args.amin or args.atime:
+        init += ['atime INT64']
+
+    if args.cmin or args.ctime:
+        init += ['ctime INT64']
+
+    if args.mmin or args.mtime or args.newer:
+        init += ['mtime INT64']
+
+    if args.empty or args.size or args.smallest or args.largest:
+        init += ['size INT64']
 
-    select = []
-    for columns in args.select:
-        select += columns
+    if args.executable or args.readable or args.writable:
+        init += ['mode INT64']
 
-    entries += [','.join(select)]
+    if args.uid or args.user:
+        init += ['uid INT64']
 
-    return tsummary, summary, entries
+    if args.gid or args.group:
+        init += ['gid INT64']
+
+    if args.links:
+        init += ['nlinks INT64']
+
+    if args.type:
+        init += ['type TEXT']
+
+    return init
+
+def build_select(args):
+    '''Build the SELECT clause'''
+
+    select = ['(path() || "/" || name) as name', 'type']
+
+    return select
 
 def build_where(args, root_uid = 0, root_gid = 0):
-    '''Build the WHERE portion of the queries for the treesummary, summary, and entries tables.'''
-    tsummary = []
-    summary  = []
```

Comment:
Still need to init summary if you are going to continue using it down below. Not sure it is necessary in the size args == 1 case, but will defer to you.

Reply:
The initial thought was that the different queries would have very different queries, but now it seems that most columns are shared, and the differences are minor enough to be hard coded into the query generation input.